hikari/snowflakes.py

```diff
@@ -136,6 +141,28 @@ def __eq__(self, other: typing.Any) -> bool:
         return type(self) is type(other) and self.id == other.id
 
 
+def calculate_shard_id(
+    app_or_count: typing.Union[traits.ShardAware, int, str], guild: SnowflakeishOr["guilds.PartialGuild"]
+) -> int:
+    """Calculate the shard ID for a guild based on it's shard aware app or shard count.
+
+    Parameters
+    ----------
+    app_or_count : hikari.traits.ShardAware or builtins.int or builtins.str
+        The shard aware app of the current application or the integer count of
```

Comment:
why str?

Reply:
also typing.union pls :)

Reply:
As with snowflake that's just allowing a string for an integer value

Reply:
Why would this ever be a string? We store this globally as an int as soon as we receive it and never pass it back to the API explicitly as a string. It won't ever be more than 2^32 either unless you have majorly done something wrong, so there is no need to complicate this further really.

Reply:
it shouldn't need a special case to justify arbitrarily allowing a string for an integer value in some places but not others

Reply:
If "the API requires it" is classed as an unjustifiable special case, then sure.

Shard IDs are not snowflakes, they are 32 bit integers (maybe even 16 bit integers). Snowflakes are always exposed as 64 bit values which break IEEE double precision floating point literals, which is why the API allows it. 

Since Hikari allows you to directly interface with the REST or gateway API as you see fit, it makes sense to allow both formats. However, the gateway never provides these as strings, nothing uses them as strings, so all it does is allow more than one way to specify the ID here without any benefit (in fact, it just adds a tiny amount of overhead instead potentially).

Up to you though I guess.

Reply:
> If "the API requires it" is classed as an unjustifiable special case, then sure.
isn't the whole point of an api wrapper to obfuscate off weird behaviour like that?

Reply:
If that is the case, that is an argument for _not_ adding this.

The only reason it was added in the first place was because originally we had the REST component only allowing strings for IDs and the higher level wrapper allowing anything else.

I am more than happy to support completely removing string support for IDs as well.

Reply:
I'll remove int from that then

Reply:
uhm...  keep int, it is a shard ID!

Reply:
Meant to say "str" there lol, sry